tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.Reporter.cs

```diff
@@ -93,6 +94,11 @@ public void Report(string triggerData, ulong aggregatedTotalRuntime, ulong aggre
         _localRootSpan.SetMetric(Metrics.AppSecWafAndBindingsDuration, aggregatedTotalRuntimeWithBindings);
         var headers = _httpTransport.GetRequestHeaders();
         AddHeaderTags(_localRootSpan, headers, RequestHeaders, SpanContextPropagator.HttpRequestHeadersTagPrefix);
+
+        if (status is not null)
+        {
+            _localRootSpan.SetHttpStatusCode(status.Value, isServer: true, Tracer.Instance.Settings);
```

Comment:
would it be possible here to create the function `_httpTransport.GetResponseStatusCode` and call it here directly?

Reply:
I don't really want to make this change, since most of the time this method isn't responsible for setting the status code. It's only need for WebApi when we throw, and I'm not sure when the transport gets notified on the status code change in this case.

Perhaps it's better to move the setting of the status code outside of this method for the WebApi case?

Reply:
ha ok, yes in this case, the current status code would not be 302 yet as reporting happens before the throw of course.
It would make sense to move it oustide as it's just for the WebApi case, so that it's not confusing later, we might think we always need to set it.. perhaps in `MakeReportingFunction` after `Report` is called?